### PR TITLE
Linked list implementation

### PR DESCRIPTION
**Pull request by DrCaH4ec**
Have done task №4. Implemented own class in sysfs and linked list.

### CODE REVIEW

**Review by ekovalyov — CHANGES_REQUESTED** [review]
Please check code style with checkpatch.pl

File: 04_basic_struct/task_04.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+
+/* sysfs store() method. Calls the store() method corresponding to the individual sysfs file */
+static ssize_t hello_store(struct class *class, struct class_attribute *attr, const char *buf, size_t count)
+{
+    strncpy(text_buf, buf, count);
+    text_buf[count] = '\0';
+
+    tmp = kzalloc(sizeof(*tmp), GFP_KERNEL);
+    strcpy(tmp->text, text_buf);
```
Comment (ekovalyov) [inline 04_basic_struct/task_04.c:63]: It is possible to use kstrdup instead of alloc and copy.

File: 04_basic_struct/task_04.c [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+void sysfs_mod_cleanup(void)
+{
+    struct list_head *pos = NULL;
+
+    list_for_each(pos, &my_list) {
+            struct _tmp_head *tmp = list_entry(pos, struct _tmp_head, head);
+
+            list_del(pos);
+            kfree(tmp);
```
Comment (ekovalyov) [inline 04_basic_struct/task_04.c]: two different indentation in same function.

Comment (DrCaH4ec) [reply 04_basic_struct/task_04.c]: But this indentation for the loop _list_for_each_. In another way, I don't understand your comment.

Comment (ekovalyov) [reply 04_basic_struct/task_04.c]: Line 96 uses 4 spaces for indentation.
Line 99 uses 8 spaces for indentation.

**Review by ekovalyov — CHANGES_REQUESTED** [review]
Please fix indentation.

**Pull request merged**